### PR TITLE
[New Migration Workflow] New Assessment Summary Card

### PR DESCRIPTION
**Pull request by krritik**
This contains code for the new assessment summary card.

UI Mocks - https://www.figma.com/file/EC2tJwlwy4RveRYhhGiHP4/Proposal-for-UI-improvement---SQL-Migration-ADS?type=whiteboard&node-id=0-1&t=Q3e7eXA4TNRvAzde-0  

A sample output using the same card - 
image

### CODE REVIEW

File: extensions/sql-migration/src/wizard/skuRecommendation/assessmentSummaryCard.ts [hunk 1]
```diff
@@ -0,0 +1,422 @@
...
+			CSSStyles: {
+				'height': '394px',
+				'width': '244px',
+				'text-align': 'center',
+				'display': 'flex',
+				'flex-direction': 'column',
+				'justify-content': 'center',
+				'box-shadow': '0px 1px 4px rgba(0, 0, 0, 0.13)',
+				'border-radius': '2px',
+				// 'margin-top': '16px',
```
Comment (stuti149) [inline extensions/sql-migration/src/wizard/skuRecommendation/assessmentSummaryCard.ts:45]: remove commented css if not needed. Define css in styles.ts file

File: extensions/sql-migration/src/wizard/skuRecommendation/assessmentSummaryCard.ts [hunk 2]
```diff
@@ -0,0 +1,422 @@
...
+			value: "",
+			width: 147,
+			height: 40,
+			CSSStyles: {
+				'font-size': '14px',
+				'line-height': '20px',
+				'font-weight': '600',
+				'margin': '0px',
+				'padding-top': '5px',
+				// 'padding-left': '44px'
```
Comment (stuti149) [inline extensions/sql-migration/src/wizard/skuRecommendation/assessmentSummaryCard.ts:76]: same here

File: extensions/sql-migration/src/wizard/skuRecommendation/assessmentSummaryCard.ts [hunk 3]
```diff
@@ -0,0 +1,422 @@
...
+					...styles.ASSESSMENT_SUMMARY_CARD_CSS,
+					'color': '#57A300',
+				};
+				this._readyText = value;
+				break;
+			case AssessmentResultType.NEEDS_REVIEW:
+				label.value = constants.NEEDS_REVIEW;
+				value.CSSStyles = {
+					...styles.ASSESSMENT_SUMMARY_CARD_CSS,
+					'color': '#DB7500',
```
Comment (stuti149) [inline extensions/sql-migration/src/wizard/skuRecommendation/assessmentSummaryCard.ts:278]: colors are defines in helper.ts file, you can use from there or define your new color code there

File: extensions/sql-migration/src/wizard/skuRecommendation/assessmentSummaryCard.ts [hunk 4]
```diff
@@ -0,0 +1,422 @@
...
+	}
+
+	// TODO - Check this later, if we need to handle this separately.
+	public dispose(): void {
+		this._disposables.forEach(
+			d => { try { d.dispose(); } catch { } });
+	}
+}
+
+export enum AssessmentResultType {
```
Comment (stuti149) [inline extensions/sql-migration/src/wizard/skuRecommendation/assessmentSummaryCard.ts:406]: Let's define this enum where all other enums are defined

File: extensions/sql-migration/src/constants/strings.ts [hunk 5]
```diff
@@ -137,9 +139,9 @@ export const REFRESH_ASSESSMENT_BUTTON_LABEL = localize('sql.migration.refresh.a
 export const SKU_RECOMMENDATION_CHOOSE_A_TARGET = localize('sql.migration.wizard.sku.choose_a_target', "Choose your Azure SQL target");
 export const SKU_RECOMMENDATION_CHOOSE_A_TARGET_HELP = localize('sql.migration.wizard.sku.choose_a_target.help', "Not sure which Azure SQL target is right for you? Learn more");
 
-export const SKU_RECOMMENDATION_MI_CARD_TEXT = localize('sql.migration.sku.mi.card.title', "Azure SQL Managed Instance");
-export const SKU_RECOMMENDATION_SQLDB_CARD_TEXT = localize('sql.migration.sku.sqldb.card.title', "Azure SQL Database");
-export const SKU_RECOMMENDATION_VM_CARD_TEXT = localize('sql.migration.sku.vm.card.title', "SQL Server on Azure Virtual Machine");
+export const SKU_RECOMMENDATION_MI_CARD_TEXT = localize('sql.migration.sku.mi.card.title', "Azure SQL Managed Instance (PAAS)");
```
Comment (Ramudaykumar) [inline extensions/sql-migration/src/constants/strings.ts:142]: Just Check with PM whether they want "PAAS" or "PaaS" and Same for "IAAS" "IaaS".

**Review by Ramudaykumar — COMMENTED** [review]
Add required comments in different sections, which will help understanding code to others.

File: extensions/sql-migration/src/constants/strings.ts (hunk 5 above)
Comment (krritik) [reply extensions/sql-migration/src/constants/strings.ts:142]: Done. It should be IaaS and PaaS

**Comment by krritik** [comment]
> Add required comments in different sections, which will help understanding code to others.

Done

File: extensions/sql-migration/src/wizard/skuRecommendation/assessmentSummaryCard.ts (hunk 4 above)
Comment (krritik) [reply extensions/sql-migration/src/wizard/skuRecommendation/assessmentSummaryCard.ts:406]: This enum is specific to this file. I saw similar for other files as well. Only enums common to different files are in utils.ts

File: extensions/sql-migration/src/constants/strings.ts [hunk 6]
```diff
@@ -87,6 +87,7 @@ export const ASSESSMENT_RESULTS_PAGE_HEADER = localize('sql.migration.assessment
 export const DATABASES_ASSESSED_LABEL = localize('sql.migration.database.assessed.label', "Database(s) assessed");
 export const MIGRATION_TIME_LABEL = localize('sql.migration.migration.time.label', "Ready for migration");
 export const ASSESSMENT_FINDINGS_LABEL = localize('sql.migration.assessment.findings.label', "Assessment findings");
+export const MIGRATION_READINESS_LABEL = localize('sql.migration.migration.readiness.label', "Migration Readiness");
```
Comment (Ramudaykumar) [inline extensions/sql-migration/src/constants/strings.ts:90]: Text should be sentence casing.

File: extensions/sql-migration/src/wizard/skuRecommendation/assessmentSummaryCard.ts [hunk 7]
```diff
@@ -0,0 +1,427 @@
...
+				container.addItem(this._vmRecommendedConfigurationText);
+		}
+
+		return container;
+	}
+
+	// Used to update the summary of assessment card later after its initialization.
+	// TODO - We can remove dummayData input later and use a some other way to pass the values. Once we start implementing the whole page.
+	public async updateContent(dummyData: DummyData) {
+		await this._assessmentResultText.updateProperties({ "value": constants.ASSESSED_DBS(dummyData.assessmentResult) });
```
Comment (brian-harris) [inline extensions/sql-migration/src/wizard/skuRecommendation/assessmentSummaryCard.ts:391]: recommend using the .value property directly instead unless you need it to be asynchronously updated.
as this._assessmentResultText.value("new value")

**Pull request merged**